v2/logging.go

```diff
@@ -1,45 +1,23 @@
 package shuttle
 
 import (
-	"context"
-	"fmt"
-	"os"
-	"time"
+	"io"
+	"log/slog"
 )
 
-type Logger interface {
-	Info(s string)
-	Warn(s string)
-	Error(s string)
-}
-
-// SetLoggerFunc sets the function to be used to acquire a logger when go-shuttle logs.
-func SetLoggerFunc(fn func(ctx context.Context) Logger) {
```

Comment:
We can deprecate this and keep it around.
maybe even **move** the func into a "deprecated.go" file